### PR TITLE
Fix #12483 Incorrect specification of PrimFloat.leb

### LINKED ISSUE #12483: Incorrect specification of PrimFloat.leb
```Coq
Require Import Floats.

Goal False.
Proof.
cut (false = true).
{ intro H; discriminate H. }
change false with (1 <= 0)%float.
rewrite leb_spec.
reflexivity.
Qed.
```

Coq Version: 8.11

The issue comes from the specification of PrimFloat.leb:
```Coq
  Definition SFleb f1 f2 :=
    match SFcompare f1 f2 with
    | Some Le => true
    | _ => false
    end.
```
Since `Le` is not a constructor, it becomes an unused variable (without warning).

The correct specification should replace `Le` by `Lt | Eq` (I'm preparing a PR).

### PR DESCRIPTION
**Pull request by proux01**
**Kind:** bug fix

Fixes: #12483 

- [x] Added / updated test-suite
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by proux01** [comment]
@Zimmi48 I put the 8.12 milestone as it fixes a proof of False, but feel free to change this.

**Review by silene — APPROVED** [review]

**Review by erikmd — APPROVED** [review]

**Comment by erikmd** [comment]
regarding the handling of such issues, I believe one needs to update [this document](https://github.com/coq/coq/blob/master/dev/doc/critical-bugs) as well? (but maybe this might be done in a later PR if the timeline of 8.12+beta1 is too close)

File: dev/doc/critical-bugs [hunk 1]
```diff
@@ -347,6 +347,16 @@ Conflicts with axioms in library
   GH issue number: none
   risk:
 
+  component: primitive floating-points
+  summary: Incorrect specification of PrimFloat.leb
+  introduced: 8.11
+  impacted released versions: 8.11.0, 8.11.1
```
Comment (silene) [inline dev/doc/critical-bugs:353]: Also 8.11.2.

Comment (proux01) [reply dev/doc/critical-bugs:353]: Indeed

Comment (Zimmi48) [reply dev/doc/critical-bugs:353]: You didn't push any change @proux01, despite having marked this as resolved.

Comment (proux01) [reply dev/doc/critical-bugs]: Oups, thanks, here it is.

**Comment by proux01** [comment]
CI green, this will need an assignee.

**Comment by Zimmi48** [comment]
@silene Do you want to take care of merging this?

**Pull request merged**